pipenv/utils.py

```diff
@@ -240,54 +241,43 @@ class PipCommand(basecommand.Command):
         req_dir = TemporaryDirectory(suffix='-requirements', prefix='pipenv-')
         cleanup_req_dir = True
     for dep in deps:
-        if dep:
-            if dep.startswith('-e '):
-                constraint = req.InstallRequirement.from_editable(
-                    dep[len('-e '):]
-                )
-            else:
-                fd, t = tempfile.mkstemp(
-                    prefix='pipenv-', suffix='-requirement.txt', dir=req_dir.name
-                )
-                with os.fdopen(fd, 'w') as f:
-                    f.write(dep)
-                constraint = [
-                    c for c in req.parse_requirements(t, session=pip_requests)
-                ][
-                    0
-                ]
-            # extra_constraints = []
-            if ' -i ' in dep:
-                index_lookup[constraint.name] = project.get_source(
-                    url=dep.split(' -i ')[1]
-                ).get(
-                    'name'
-                )
-            if constraint.markers:
-                markers_lookup[constraint.name] = str(
-                    constraint.markers
-                ).replace(
-                    '"', "'"
-                )
-            constraints.append(constraint)
+        if not dep:
+            continue
+        url = None
+        if ' -i ' in dep:
+            dep, url = dep.split(' -i ')
+        req = Requirement.from_line(dep)
+        constraints.append(req.as_line())
```

Comment:
What does this `from_line` → `to_line` combination do? Can’t `dep` be added directly into constraints?